### ISSUE TITLE
Array-value default for Postgres UUID Array type fails in 5.1.2+ migrations

### ISSUE THREAD

**Issue by booleanbetrayal** [rails/rails#30539]
### Steps to reproduce

(Guidelines for creating a bug report are [available
here](http://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#creating-a-bug-report))

### Expected behavior

During a rails migration, the following schema change definition should continue to work 5.1.2+ with PostgreSQL 9.x:

`add_column :some_table, :some_column, :uuid, array: true, default: []`

### Actual behavior

The statement fails with:
```
no implicit conversion of Array into String
activesupport-5.1.2/lib/active_support/core_ext/regexp.rb:7:in `match'
activesupport-5.1.2/lib/active_support/core_ext/regexp.rb:7:in `match?'
activerecord-5.1.2/lib/active_record/connection_adapters/postgresql/quoting.rb:65:in `quote_default_expression'
...
```
This worked previously in 5.1.1. A workaround is available using the PostgreSQL string array literal format:

`add_column :some_table, :some_column, :uuid, array: true, default: '{}'`

### System configuration
**Rails version**:
5.1.2 / 5.1.3

**Ruby version**:
2.3.x

**Comment by booleanbetrayal** [comment]
you're too fast @kamipo  --- I was still finishing up the minitest for this.  =]

**Comment by cdesch** [comment]
@booleanbetrayal How do you get this to work with Rails 5.1.4? I see @kamipo pull request to get it into Rails 5.2 ... which is awesome, but how do I handle this now?

**Comment by booleanbetrayal** [comment]
@cdesch - if I recall, the Postgres literal worked for this: `default: '{}'`

**Comment by cdesch** [comment]
Thanks @booleanbetrayal - I'll give that a shot on my next try using the migrations. For the interim I just use SQL `execute` in the migration:

  ```
def up
    execute "ALTER TABLE requests ADD COLUMN examples uuid[] NOT null DEFAULT '{}'"
    execute "ALTER TABLE requests ADD COLUMN types uuid[] NOT null DEFAULT '{}'"
    execute "ALTER TABLE requests ADD COLUMN parts uuid[] NOT null DEFAULT '{}'"
  end

```

**Comment by likeuwill** [comment]
That was implemented in 5.1.5 ;(

**Comment by cdesch** [comment]
Thanks @likeuwill! Can confirm it works.
